lib/mozart_fetcher/router.ex

```diff
@@ -47,7 +47,7 @@ defmodule MozartFetcher.Router do
       {:error} ->
         ExMetrics.increment("error.components.decode")
         Stump.log(:error, %{message: "Failed to decode components into list"})
-        {:error}
+        :error
```

Comment:
What is the reason behind this change? This is inconsistent with the `{:ok, _}` and {:error, _} cases in this and other modules. Could return a {error, exception} that provide the error messages for the preceding `send_response/2`, rather than for it to be hard-coded as below:

```
 def send_response(:error, conn) do
    conn
    |> put_resp_content_type("application/json")
    |> send_resp(500, "Internal Server Error")
  end
```
To 

```
 def send_response({:error, exception}, conn) do
    conn
    |> put_resp_content_type("application/json")
    |> send_resp(exception.status, exception.message)
  end
```

Could reuse existing exceptions (500 from HTTP clients) for other library or define one with `defexception`.

Reply:
I have followed the approach you have suggested, we dont really update this application much though so I dont think the exception status/message is going to be used massively but I think the main thing to gain from it is setting a good example for others who may look through our code base, thanks for pointing it out!

Reply:
Thanks Joe, that's a pragmatic solution. A good starting point for more elaborated error handling in the future if needs be.